api/graphql/schema.graphql

```diff
@@ -1716,6 +1741,10 @@ type Mutation {
       model: "PoolCandidate"
       injectArgs: true
     )
+  updatePoolCandidateClaimVerification(
```

Comment:
I forgot to add this AC to the backend when I split the issue:
> The expiration date is Required for Priority status (when it is shown) but is NEVER required for Veteran status.

Could you add this to backend validation?  A validator for UpdatePoolCandidateClaimVerificationInput might be a good place for this.

Reply:
I'm not sure I follow.
Priority expired is required but when? I don't follow what "when it is shown" means. What `ClaimResult` values does that map to?

Reply:
When priority is ACCEPTED then expiration date is required.   When veteran is ACCEPTED then expiration date is optional.  Hope that helps - let me know if you want to chat about it.

Reply:
Ah thanks, that explains it nicely
Updated in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10434/commits/821dd17b9c90f834f6ceb132184183a198f1ffb1

Reply:
Nice, this rule is good.  I wonder if we want to prevent setting the expiry dates if the statuses are anything other than ACCEPTED.  Is it bad to set an expiry date if the status is REJECTED or UNVERIFIED?  :thinking:

Reply:
Gray said only ACCEPTED gets an expiration

Reply:
Okay, done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10434/commits/9da6401062d6511b983f4ccd128beb8122fafa43